src/app/institutional/pages/institutional/institutional.component.ts

```diff
@@ -0,0 +1,181 @@
+import { Component, Inject, OnInit } from '@angular/core'
+import { WINDOW } from '../../../cdk/window'
+import { FormControl, FormGroup, Validators } from '@angular/forms'
+import { Observable } from 'rxjs'
+import { map, startWith, take } from 'rxjs/operators'
+import { DiscoService } from '../../../core/disco/disco.service'
+import { Institutional } from '../../../types/institutional.endpoint'
+import { CookieService } from 'ngx-cookie-service'
+import { environment } from '../../../../environments/environment.local'
+import { InstitutionValidator } from '../../../shared/validators/institution/institution.validator'
+
+@Component({
+  selector: 'app-institutional',
+  templateUrl: './institutional.component.html',
+  styleUrls: ['./institutional.component.scss'],
+  host: { class: 'container' },
+  preserveWhitespaces: true,
+})
+export class InstitutionalComponent implements OnInit {
+  loading = false
+  options: any[]
+  retrieveAllFiltered: any[]
+  filteredOptions: Observable<string[]>
+  institution: Institutional
+  entityID: any
+  logoInstitution: any
+  cookieExpirationTime = 730
+  numberOfInstitutionsToDisplay = 20
+
```

Comment:
cookieExpirationTime  and numberOfInstitutionsToDisplay would be better on the environment files

Reply:
Please add names that are descriptive enough for the environment file for instance

INSTITUTIONAL_AUTOCOMPLETE_COOKIE_DAYS_TO_EXPIRE
and 
INSTITUTIONAL_AUTOCOMPLETE_DISPLAY_AMOUNT

Reply:
Ok, I will add them to the environment files